### DISCUSSION TITLE
Swap two input fields (react-select)

### DISCUSSION THREAD

**Discussion by teifler** [JedWatson/react-select#5124] [Q&A] [answered]
Hey fellow hackers, 

As you can see below I have two react select input fields and on the click on the button I want to swap the two input fields so that To swaps with From. What is the best way to manipulate the input field to do that? Can I save it somehow and swap it out with useRef for Example?

<img width="353" alt="image" src="https://user-images.githubusercontent.com/10044263/160005830-1b479ad9-0463-4000-96fb-644a9ca6d83a.png">

Any help is much appreciated,
Tim

**Comment by manjushsh** [comment, marked as answer]
@teifler Something like this?
```
  const [option, updateOption] = useState(null);
  const [secondOption, updateSecondOption] = useState(null);
  
  const onSwap = () => {
    const tempStore = secondOption;
    updateSecondOption(option);
    updateOption(tempStore);
    // Calulation logic...
  };
```

Component:
```
const SimpleSelect = ({
  option,
  updateOption,
  secondOption,
  updateSecondOption,
  onSwap
}) => {
  const options = [
    { value: "121", label: "121" },
    { value: "122", label: "122" },
    { value: "123", label: "123" }
  ];

  return (
    <div className="App">
      <h3>Second React Select</h3>
      <Select
        // menuIsOpen={true}
        value={secondOption}
        onChange={updateSecondOption}
        options={options}
      />
      <br />
      <button className="btn btn-primary" onClick={onSwap}>
        Swap
      </button>
      <br />
      <br />
      <Select
        // menuIsOpen={true}
        value={option}
        onChange={updateOption}
        options={options}
      />
    </div>
  );
};
```
[Code Sandbox](https://codesandbox.io/s/affectionate-browser-nkff64?file=/src/App.js)

**Reply by teifler** [reply]
Thank you very much! Thats exactly what I need but I dont understand the codepen complelty. Why are there two returns in the codepen? I struggle with that since a couple of hours now -.-

Below is my code where I need to add the functionality.
https://github.com/teifler/capstone-project/pull/29/commits/eb3743283d71421d25669409d017849a6fdc5135#diff-55a5c99aef9d08f686c49b7d705cc616f4add4722c1ba86869dc3917a5424a48

**Reply by manjushsh** [reply]
refer above code.. I added code sandbox just to show running demo. First return is for main component and second return for `SimpleSelect` component.

**Reply by teifler** [reply]
Cheers I fixed the issue thank you very much for your help! :)
